hdk/jack_analyzer/parser.py

```diff
@@ -0,0 +1,333 @@
+from typing import Iterable, Iterator
+
+from hdk.jack_analyzer.syntax import Expression, UnaryOpTerm, ConstantTermType, SubroutineCall, Expressions, \
+    LetStatement, \
+    DoStatement, ReturnStatement, WhileStatement, IfStatement, Statements, VarDeclaration, ClassVarDeclaration, ParameterList, SubroutineBody, \
+    SubroutineDeclaration, Class, ConstantTerm, SubroutineCallTerm, ExpressionTerm, Parameter, VarTerm, Term
+from hdk.jack_analyzer.tokenizer import Token, tokenize, TokenType
+
+
+class TokensIterator(Iterator[Token]):
+    def __init__(self, tokens: Iterable[Token]):
+        self._tokens_iterator = iter(tokens)
+        self._look_ahead: Token | None = None
+
+    def peek(self) -> Token:
+        if self._look_ahead is None:
+            self._look_ahead = next(self._tokens_iterator)
+        return self._look_ahead
+
+    def __next__(self) -> Token:
+        if self._look_ahead:
+            value = self._look_ahead
+        else:
+            value = next(self._tokens_iterator)
+        self._look_ahead = None
+        return value
+
+
+def parse_var(tokens: TokensIterator) -> VarDeclaration:
+    """
+    >>> expression_text = 'int i, j;'
+    >>> expression_tokens = tokenize(expression_text)
+    >>> parse_var(TokensIterator(expression_tokens))
+    """
+    type_ = next(tokens).value
+    var_names_list = [next(tokens).value]
```

Comment:
var_names_list -> names
user construction while next(token).value == ",": to reduce the number of next()